tests/rbd_mirror/rbd_mirror_reconfigure_secondary_cluster.py

```diff
@@ -0,0 +1,114 @@
+    conf and keyring files
+
+    Test case flows:
+    1) After site-b failure, bring up new cluster for secondary
+    2) Create a pool with same name as secondary pool
+    3) Perform the mirroring bootstrap for cluster peers
+    4) copy and import bootstrap token to peer cluster
+    5) verify peer cluster got added successfully after failover
+    6) verify all the images from secondary mirrored to primary
+"""
+import datetime
+import time
+
+from tests.rbd.rbd_utils import Rbd
+from tests.rbd_mirror import rbd_mirror_utils as rbdmirror
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(**kw):
+    """
+    Verification of rbd mirroring failover scenario on secondary site,
+    This module configures the mirroring to newly created cluster.
+
+    Args:
+        kw: test data
+        Example::
+            config:
+                poolname: pool_failback
+    Returns:
+        int: The return value - 0 for success, 1 for failure
+    """
+    try:
+        log.info("Running test CEPH-9477")
+
+        config = kw.get("config")
+        poolname = config.get("poolname")
+
+        mirror3 = rbdmirror.RbdMirror(
+            kw.get("ceph_cluster_dict").get("ceph-rbd3"), config
+        )
+
+        mirror1 = rbdmirror.RbdMirror(
+            kw.get("ceph_cluster_dict").get("ceph-rbd1"), config
+        )
+
+        rbd1, _, rbd3 = [
+            Rbd(**kw, req_cname=cluster_name)
+            for cluster_name in kw["ceph_cluster_dict"].keys()
+        ]
+
+        kw["peer_mode"] = kw.get("peer_mode", "bootstrap")
+        kw["rbd_client"] = kw.get("rbd_client", "client.admin")
+        kw["failback"] = True
+
+        log.info("Re-establishing mirror with newly created cluster as Primary")
+
+        # # remove the failed cluster from mirror peers
+        mirror1.peer_remove(poolname=poolname)
+
+        # # For newly added cluster, creating pool and configuring two-way mirror
+        mirror3.create_pool(poolname=poolname)
+
+        # # set custom_cluster_order according to the cluster to configure mirroring
+        kw["custom_cluster_order"] = "1,2"
+
+        mirror1.config_mirror(
+            mirror3,
+            poolname=poolname,
+            mode="image",
+            start_rbd_mirror_primary=True,
+            **kw,
+        )
+
+        # # Schedeluing the snapshots for image reflection
+        mirror1.mirror_snapshot_schedule_add(poolname=poolname)
+
+        # # wait till image gets reflected to secondary
+        end_time = datetime.datetime.now() + datetime.timedelta(seconds=300)
+        while end_time > datetime.datetime.now():
+            mirror3_image_list = rbd3.list_images(poolname)
+            mirror1_image_list = rbd1.list_images(poolname)
+            if set(mirror3_image_list) == set(mirror1_image_list):
+                log.info(
+                    "After failback, secondary cluster images successfuly mirrored to primary cluster"
```

Comment:
The cluster name needs to be reversed here.

Reply:
done.